### DISCUSSION TITLE
Can anyone reccomend some youtube tutorials on how to do this?

### DISCUSSION THREAD

**Discussion by KInderhees** [EtoTen/stm32_can_sim#3] [General]
Can anyone recommend some YouTube tutorials on how to do this? im a newb for programming stm32's and dont understand how they communicate with the VStudio and what frameworks i need to make this work. 

Ive programmed Arduinos in the past but im lost at how to load the programs onto the stm. I would really like to do this to put a sync screen in a portable boom box i, building. If anyone is able to point me in the right direction that would be really appreciated.

**Comment by EtoTen** [comment]
Just install vscode and install platformio (framework) using the built in vscode "app store". Then open my project, it should download the proper dependencies let you then compile/upload/monitor to a USB connected stm32 bluepill dev board. The dependencies include the stm32 flash utility which all auto-magically runs and works to let you upload to the board.

**Reply by Unknown-iOS** [reply]
Hey Nikolai I am in need of help. 
I do sycn 3 conversion and I am needing help on how to program the stm32. English is not my forte. But Anything helps.

**Comment by Hacki12** [comment]
Hey Guys, 
First, Thank You EtoTen for your Work here. That`s realy a great project!
But, i have problems with the programming. I have installed VScode. Then i have installed ( in vscode ) the platformio ide "application".
i have experience with raspberry pi and arduino but not with stm32 and vscode. 
how do i get your project in vscode and how do i upload it to the stm32?
excuse my stupidity! Can you help with a tutorial by text or video? For you it`s a few minutes, for me days :)
Hacki

**Reply by Unknown-iOS** [reply]
Any help or Youtube Videos

**Reply by EtoTen** [reply]
> Any help or Youtube Videos

how about: https://stm32-base.org/guides/platformio.html

**Comment by EtoTen** [comment]
Also sorry I missed the OP, hopefully you were able to get it running. Basically you install vscode, then in vscode, install platformio, then open my cloned project in vscode, it should then show the platformio view. From there you would do the "upload" option, under the device specific project in platformio.

If the board does not upload, then it could be a myriad of issues, but I would first try and comment out "upload_flags = -c set CPUTAPID 0x2ba01477" in platformio.ini. This is needed to trick the utility to accept fake STM32 chips, but will make it not upload to the real board.
